hibernate-orm-panache-kotlin-quickstart fails to build native after "Unconditionally enable complete reflection type changes in quarkus"

### Describe the bug

hibernate-orm-panache-kotlin-quickstart fails to build native after "Unconditionally enable complete reflection type changes in quarkus"

Failure of QS: https://github.com/quarkusio/quarkus/issues/6588#issuecomment-3641099729

```
Build resources:
 - 11.80GB of memory (75.6% of 15.62GB system memory, determined at start)
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[2/8] Performing analysis...  []                                                                         (9.7s @ 0.38GB)
    3,318 reachable types   (57.0% of    5,821 total)
    2,715 reachable fields  (38.7% of    7,024 total)
    9,224 reachable methods (24.5% of   37,597 total)
    1,790 types,   788 fields, and 7,670 methods registered for reflection

Error: Class initialization of org.jboss.narayana.jta.jms.ConnectionManager failed. This error is reported at image build time because class org.jboss.narayana.jta.jms.ConnectionManager is registered for linking at image build time by command line and command line. Use the option 

    '--initialize-at-run-time=org.jboss.narayana.jta.jms.ConnectionManager'

 to explicitly request initialization of this class at run time.
Caused by: java.lang.NoClassDefFoundError: jakarta/jms/JMSException
	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized0(Native Method)
	at java.base/jdk.internal.misc.Unsafe.ensureClassInitialized(Unsafe.java:1160)
	at org.graalvm.nativeimage.builder/
```

After reverting https://github.com/quarkusio/quarkus/pull/51432 and https://github.com/quarkusio/quarkus/pull/51473 I get native build working again, cc @jerboaa 


### Expected behavior

Native image build ends successfully.

### Actual behavior

Native image build fails

### How to Reproduce?

* `git clone https://github.com/quarkusio/quarkus-quickstarts -b development`
* `sdk use java 23.1.9.r21-mandrel`
* `mvn clean package -DskipTests -f quarkus-quickstarts/hibernate-orm-panache-kotlin-quickstart -Dnative`

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Mandrel 23.1

### Mandrel or GraalVM version (if different from Java)

Mandrel 23.1

### Quarkus version or git rev

Quarkus main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveHierarchyBuildItem.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveHierarchyBuildItem.java
@@ -360,7 +360,8 @@ public final class ReflectiveHierarchyBuildItem extends MultiBuildItem {
         private static final List<String> DEFAULT_IGNORED_PACKAGES = Arrays.asList("java.", "io.reactivex.",
                 "org.reactivestreams.", "org.slf4j.", "jakarta.", "jakarta.json.",
                 "javax.net.ssl.", "javax.xml.", "javax.management.", "reactor.core.",
-                "com.fasterxml.jackson.databind.", "io.vertx.core.json.", "kotlin.");
+                "com.fasterxml.jackson.databind.", "io.vertx.core.json.", "kotlin.",
+                "javax.sql.", "javax.naming.");
         // if this gets more complicated we will need to move to some tree like structure
         static final Set<String> ALLOWED_FROM_IGNORED_PACKAGES = new HashSet<>(
                 Arrays.asList("java.math.BigDecimal", "java.math.BigInteger"));

